Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -65,27 +69,33 @@ void MatchTOF::run()
   LOGF(INFO, "Timing prepareTOFCluster: Cpu: %.3e s Real: %.3e s in %d slots", mTimerTot.CpuTime(), mTimerTot.RealTime(), mTimerTot.Counter() - 1);
   mTimerTot.Start();
 
+  for (int i = 0; i < trkType::SIZE; i++) {
+    mNumOfTracks[i] = 0;
+    mMatchedTracks[i].clear();
+    mTracksWork[i].clear();
+    mOutTOFLabels[i].clear();
+  }
+
   if (mIsworkflowON) {
     LOG(DEBUG) << "Number of entries in track tree = " << mCurrTracksTreeEntry;
 
-    if (mIsITSused) {
+    if (mIsITSTPCused) {
```

Comment:
We will need to cover the combinations of `{TPC and TPC-TRD}` seeds in "strict" mode and `{TPC, ITS-TPC, ITS-TPC-TRD, TPC,TRD}` for "relaxed" mode. And this should account for the redundancy: i.e. if TPC track is already used in some global track, it should not be considered anymore.
All this is naturally done via RecoContainer, which also takes care of different inputs and will absorb most of future changes. I would strongly suggest to switch to it. If you want, we can do this together.